src/promptflow/promptflow/executor/_tool_resolver.py

```diff
@@ -106,7 +107,12 @@ def _convert_node_literal_input_types(self, node: Node, tool: Tool, module: type
                 updated_inputs[k].value = create_image(v.value)
             elif isinstance(value_type, ValueType):
                 try:
-                    updated_inputs[k].value = value_type.parse(v.value)
+                    updated_inputs[k].value = v.value
+                    # Don't convert value type for literal node input.
+                    # We don't covert flow input/ node reference type by tool definition.
+                    # So, don't have reason to convert literal node input type.
+                    if not isinstance(updated_inputs[k].value, value_type.get_class()):
+                        logger.warning(f"Literal node input {k} is not type {value_type}, received type is {type(v)}.")
                     updated_inputs[k].value = load_multimedia_data_recursively(updated_inputs[k].value)
```

Comment:
I guess we can load multimedia before the type check?

Reply:
No, after loading multimedia, we may get `Image` for `Dict`